quickstart/app/Http/Controllers/ProductController.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Product;
+use App\Models\Category;
+use Illuminate\Support\Str;
+use Illuminate\Support\Facades\Storage;
+
+class ProductController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        try {
+            $products = Product::paginate(config('app.paginate_product'));
+        } catch (ModelNotFoundException $exception) {
+
+            return back()->withError($exception->getMessage())->withInput();
+        }
+        
+
+        return view(
+            'product.index',
+            compact('products'),
+        );
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    { 
+        try {
+            $categories = Category::all();
+        } catch (ModelNotFoundException $exception) {
+
+            return back()->withError($exception->getMessage())->withInput();
+        }
+
+        return view('product.create', compact('categories'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $product=new Product;
```

Comment:
```suggestion
        $product = new Product();
```
cách ra chứ tương tự các dòng dưới